src/classes/window/fullscreen.vala

```diff
@@ -100,19 +107,9 @@ namespace pdfpc.Window {
             this.resizable = true;
 
             if (!Options.windowed) {
-                // Move to the correct monitor
-                // This movement is done here and after mapping, to minimize flickering
-                // with window managers, which correctly handle the movement command,
-                // before the window is mapped.
-                this.move(this.screen_geometry.x, this.screen_geometry.y);
-
-                // As certain window-managers like Xfwm4 ignore movement request
-                // before the window is initially moved and set up we need to
-                // listen to this event.
-                this.size_allocate.connect(this.on_size_allocate);
-                this.configure_event.connect(this.on_configure);
-            }
-            else {
+                // start moving and fullscreening after the window was shown initially
+                this.map.connect(this.on_mapped);
```

Comment:
This should be `this.map_event`. See e.g. [here](https://blogs.gnome.org/jnelson/2010/10/13/those-realize-map-widget-signals/): `map` is a Gobject signal that is emitted once a map request has been sent to the X11 server, `map_event` is a signal that is emitted once the server acknowledged by sending a map event back to the client.